From 6b45c39db8d4ac1a5d05ae313a33acf7b2c6d3ad Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 27 May 2024 19:08:55 +0200
Subject: [PATCH 0869/1695] dts: stm32: declare RISAF instances as access
 controllers

Declare RISAFs as access controllers so that it's possible to check
firewall access rights on address ranges, and reconfigure dynamically
RISAF regions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 65f252ab9766b8df85ab12b46a0a00a7bafb114e dts: stm32: declare RISAF instances as access controllers]
Change-Id: I5fd1f040fb2185ebdf7ad95829f79e0c5d4c0353
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/382949
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi     | 8 ++++++++
 core/arch/arm/dts/stm32mp257f-dk.dts  | 4 ----
 core/arch/arm/dts/stm32mp257f-ev1.dts | 4 ----
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 4d129df01..1d5323b58 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -9,6 +9,10 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/reset/stm32mp25-resets.h>
+#include <dt-bindings/soc/stm32mp25-rif.h>
+#include <dt-bindings/soc/stm32mp25-rifsc.h>
+#include <dt-bindings/soc/stm32mp25-risab.h>
+#include <dt-bindings/soc/stm32mp25-risaf.h>
 
 / {
 	#address-cells = <2>;
@@ -615,6 +619,7 @@
 			reg = <0x420a0000 0x1000>;
 			clocks = <&rcc CK_BUS_BKPSRAM>;
 			st,mem-map = <0x0 0x42000000 0x0 0x2000>;
+			#access-controller-cells = <1>;
 		};
 
 		risaf2: risaf@420b0000 {
@@ -622,6 +627,7 @@
 			reg = <0x420b0000 0x1000>;
 			clocks = <&rcc CK_KER_OSPI1>;
 			st,mem-map = <0x0 0x60000000 0x0 0x10000000>;
+			#access-controller-cells = <1>;
 			status = "disabled";
 		};
 
@@ -630,6 +636,7 @@
 			reg = <0x420d0000 0x1000>;
 			clocks = <&rcc CK_BUS_RISAF4>;
 			st,mem-map = <0x0 0x80000000 0x1 0x00000000>;
+			#access-controller-cells = <1>;
 		};
 
 		risaf5: risaf@420e0000 {
@@ -637,6 +644,7 @@
 			reg = <0x420e0000 0x1000>;
 			clocks = <&rcc CK_BUS_PCIE>;
 			st,mem-map = <0x0 0x10000000 0x0 0x10000000>;
+			#access-controller-cells = <1>;
 			status = "disabled";
 		};
 
diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 732c03c5e..bf3e4ebb3 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -11,10 +11,6 @@
 #include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/tamper/st,stm32mp25-tamp.h>
-#include <dt-bindings/soc/stm32mp25-rif.h>
-#include <dt-bindings/soc/stm32mp25-risab.h>
-#include <dt-bindings/soc/stm32mp25-risaf.h>
-#include <dt-bindings/soc/stm32mp25-rifsc.h>
 #include "stm32mp257.dtsi"
 #include "stm32mp25xf.dtsi"
 #include "stm32mp257f-dk-ca35tdcid-rcc.dtsi"
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index dd23200d0..2d39a73c9 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -10,10 +10,6 @@
 #include <dt-bindings/mfd/st,stpmic2.h>
 #include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/soc/stm32mp-provisioning.h>
-#include <dt-bindings/soc/stm32mp25-rif.h>
-#include <dt-bindings/soc/stm32mp25-risab.h>
-#include <dt-bindings/soc/stm32mp25-risaf.h>
-#include <dt-bindings/soc/stm32mp25-rifsc.h>
 #include <dt-bindings/tamper/st,stm32mp25-tamp.h>
 #include "stm32mp257.dtsi"
 #include "stm32mp25xf.dtsi"
-- 
2.47.2

